### PR TITLE
Xenoarch hotfixes and map changes and stuff

### LINKED ISSUE #27679: Removing Fourier Transform Spectroscope or Hyperspectral Imager circuitboards nullspaces any inserted trays
Title. @DeityLink

### PR DESCRIPTION
**Pull request by DeityLink**
Fixes #27679
imageimage
imageimageimage

:cl:
* experiment: Here's a bunch of Xenoarch and Research Oupost fixes and tweaks
* bugfix: Fixed analysis trays not having their sprite update when reagents are added or removed from them.
* bugfix: Fixed the analysis machines only heating up when idle and not when scanning, regardless of the temperature of the air around them.
* bugfix: Fixed the analysis machines heating up at an absurd rate. I removed the cap on entropy so cold air in the pen where the machines are is a lot more efficient at keeping them at working temperatures. The rate at which they heat up now depends on their power consumption which has been reworked for both their idle and active states.
* experiment: This means however that the machines will overheat if you try to use them at room temperature. From now on when you arrive at the outpost and plan to do triangulation later, remember to turn on the Freezer so the pen has time to cool down. Upgrading the machines with phasic scanning modules or better however will let them complete the scan before they have time to overheat, so that's an alternative.
* tweak: The temperature gauge on the machines now goes all the way from 0K to 400K.
* tweak: Replaced the Nitrogen Pressure Tank with a Plasma Canister on top of a connector port on all maps (I used a Plasma Pressure Tank on maps where the canister couldn't be reached) which has much better specific heat and thus is much better at keeping the pen cold.
* tweak: Removed the extra Air Alarm and vents in the Anomalous Materials Lab pen since it would always trigger alarms if you started cooling. Added a Gas Meter on the floor instead that can be examined from afar to check the pen's temperature.
* bugfix: Fixed the Research Outpost Atmospherics' Gas Mixer taking 80 percent Oxygen and 20 percent Nitrogen instead of the other way around. It's been like that since forever? jesus christ.
* tweak: Lots of small changes to the box Research Outpost beside the Anomalous Materials Lab. Removed one of the SMES, a single one is enough to keep the outpost powered for almost 4 hours, and extended the hallway with an airlock on the other side of the transit tube so you can easily set up the solars in the crate next to them, which have been found after testing to keep the outpost powered for up to 11 hours if set up early. Also added a Power Monitoring Computer to the Outpost Power Substation. Lots of other small QoL changes, moves items on tables a bit, added a Newscaster, a Mousecube box, added a pair of Suspension Field Gens to the Temporary storage for old times sake. Replaced the tables with the redundant pickaxe and excavation picks with crates that xenoarchaeologists can use to easily gather their small finds, along with a table on which can be found the new revamped Xenoarchaeology manual.
* imageadd: Said manual, Out on the Dig New Edition! also features a new sprite.
* tweak: The outposts of other maps have been tweaked a bit as well, but mostly just the Anomalous Materials Lab pen, and the new Out on the Dig manual.

### CODE REVIEW

**Comment by Kharkov72** [comment]
Untitled
how

**Comment by ghost** [comment]
when u try to fix ancient shitcode but it's so cursed that it breaks github

**Comment by Gazathor** [comment]
i think this is a sign from god that we shouldnt touch the code of those before us.

**Comment by DeityLink** [comment]
Well that took me over an hour but I vanquished the evils of Github. Going to bed for real now though.

**Comment by kernikov** [comment]
New bug for you buddy.

**Comment by DeityLink** [comment]
Alright, changelog is up

File: code/modules/research/xenoarchaeology/machinery/analysis_base.dm [hunk 1]
```diff
@@ -51,67 +49,66 @@
...
 
 /obj/machinery/anomaly/process()
-	//not sure if everything needs to heat up, or just the GLPC
-	var/datum/gas_mixture/env = loc.return_air()
-	var/environmental_temp = env.temperature
+	//First we deal with the machine's task
 	if(scan_process)
-		// Shouldn't be reachable, still can't hurt.
-		if(stat & NOPOWER)
+		if (stat & (NOPOWER|BROKEN))
```
Comment (Zth--) [inline code/modules/research/xenoarchaeology/machinery/analysis_base.dm:58]: You only check if it's broken or has no power if you're scanning. I don't see any reason for that, it should also check when it's idle.

But also it doesn't return. So if it's broken or has no power then it processes the temperature changes as if it was idle/working

File: code/modules/research/xenoarchaeology/machinery/analysis_base.dm [hunk 2]
```diff
@@ -51,67 +49,66 @@
...
-			visible_message("<span class='notice'>[bicon(src)] bleets plaintively.</span>")
-
-		//show we're busy
-		if(prob(5))
-			visible_message("<span class='notice'>[bicon(src)] [pick("whirrs","chuffs","clicks")][pick(" quietly"," softly"," sadly"," excitedly"," energetically"," angrily"," plaintively")].</span>")
-
-		use_power = 2
-
+		else
+			use_power = 2
```
Comment (Zth--) [inline code/modules/research/xenoarchaeology/machinery/analysis_base.dm]: Magic numbers bad. It was not until I read line 80 that I figured that (1 == idle) and (2 == active).

Use the defines from machinery.dm

```
#define MACHINE_POWER_USE_NONE   0
#define MACHINE_POWER_USE_IDLE   1
#define MACHINE_POWER_USE_ACTIVE 2
```

File: code/modules/research/xenoarchaeology/machinery/analysis_base.dm [hunk 3]
```diff
@@ -51,67 +49,66 @@
...
 		use_power = 1
 
-	//Add 3000 joules when active.  This is about 6 degrees per tick.
-	//May need adjustment
-	if(use_power == 1)
-		var/heat_added = active_power_usage * XENOARCH_HEAT_COEFFICIENT
+	//Next, temperature management. First we automatically heat up as long as we're powered, depending on how much power we're using.
+	var/heat_added
+	switch (use_power)
+		if (1)
```
Comment (Zth--) [inline code/modules/research/xenoarchaeology/machinery/analysis_base.dm]: Ditto magic numbers

File: code/modules/research/xenoarchaeology/tools/tools_coresampler.dm [hunk 4]
```diff
@@ -50,16 +50,16 @@
 				icon_state = "sampler-full"
 
 				//put in a rock sliver
-				var/obj/item/weapon/rocksliver/R = new(src)
-				R.geological_data = geo_data
+				extracted = new(src)
```
Comment (Zth--) [inline code/modules/research/xenoarchaeology/tools/tools_coresampler.dm:59]: Hard deletes if, let's say, an explosion destroys this object before the person ejects the sample.

Create a Destroy() for this object and clear `extracted` if it exists

**Review by Zth-- — CHANGES_REQUESTED** [review]

File: code/modules/research/xenoarchaeology/machinery/analysis_base.dm (hunk 1 above)
Comment (DeityLink) [reply code/modules/research/xenoarchaeology/machinery/analysis_base.dm:58]: That specifically is intended. In the future I'm thinking of having temperature management be a machine flag so we can have the same work with tcomms machines for example. But yes, I want the temperature to change independantly of whether it's broken or powered (and whether it's broken or powered only matters if it was scanning, in which case it'll need to stop)

But now that you mention it, I'll make sure it doesn't produce any extra heat while broken/unpowered but I still want the entropy to regulate the temperature between it and the environment.

File: code/modules/research/xenoarchaeology/tools/tools_coresampler.dm (hunk 4 above)
Comment (DeityLink) [reply code/modules/research/xenoarchaeology/tools/tools_coresampler.dm:59]: done

**Review by Zth-- — APPROVED** [review]

**Comment by PJB3005** [comment]
Everything I've done to make xenoarch *less* tedious and then you get this shit forcing you to cool the analysis machines.

👎

**Comment by PJB3005** [comment]
Nevermind, on further thought I realize the xenoarch updates completely obsoleted triangulation in literally any capacity and by extension these now-unusable-due-to-the-update machines so breaking them didn't matter.

👍

**Comment by DeityLink** [comment]
Wow, that's like, not my experience playing xenoarch at all like, doing the triangulation is still the fastest and most guaranteed way to find your first large artifact. Just turn on the cooling at roundstart the same way Medbay turns on the cryo freezer and it'll be way cold enough by the time you arrive with samples.

Or if you decide to do xenoarch later into the round and R&D is done just upgrade the machine and they'll complete their analysis before they even have time to overheat at room temperature.

@PJB3005 no really what's with the passive aggressive PR grave-digging you're doing here? Not to mention we've talked about the update half a year ago when I explained to you how until now analysis distance was given in a square rather than a circle.

Xenoarch has never been less tedious, people have never played xenoarch more than nowadays, to the point where some may consider reducing the large artifact spawn chances given how common a sight they've become.

**Pull request merged**